File: conference/src/types/Conference.ts
export type Conference = {
  id: number;
  userid: number;
  title: string;
  keywords: string;
  abstract: string;
  createdon: string;
  updatedon: string;
  acronym: string;
  webpage: string;
  venue: string;
  city: string;
  startDate: string;
  endDate: string;
  researchAreas: string[];
  areaNotes: string;
  organizer: string;
  organizerWebpage: string;
  contactPhoneNumber: string;
  isAnonymizedFileRequired: boolean;
  isParticipant: boolean;
};

export type Submission = {
  id: number;
  conferenceId: number;
  authorId: number;
  authoutEmail: string;
  authorName: string;
  status: number;
  statusLabel: string;
  title: string;
  keywords: string;
  abstract: string;
  researchAreas: string[];
  createdOn: string;
  modifiedOn: string;
  isValidForReturn: boolean;
  isValidForUpdate: boolean;
  isValidForReview: boolean;
  isReviewer: boolean;
};

export type SubmissionPaper = {
  id: number;
  submissionId: number;
  fileName: string;
  base64Content: string;
  createdOn: string;
  typeLabel: string;
};

export type Comment = {
  id: number;
  submissionId: number;
  authorId: number;
  authorName: string;
  authorEmail: string;
  text: string;
  createdOn: string;
  modifiedOn: string;
  isModified: boolean;
  isAuthor: boolean;
}

export type InviteCode = {
  code: string;
  role: string;
}

File: ConferenceManager/ConferenceManager.Domain/Entities/Conference.cs
using ConferenceManager.Domain.Common;

namespace ConferenceManager.Domain.Entities
{
    public class Conference : BaseAuditableEntity
    {
        public const string ResearchAreasSeparator = "|";

        public required string Title { set; get; }

        public required string Acronym { set; get; }

        public required string Organizer { set; get; }

        public required DateTime StartDate { set; get; }

        public required DateTime EndDate { set; get; }

        public string? Keywords { set; get; }

        public string? Abstract { set; get; }

        public string? Webpage { set; get; }

        public string? Venue { set; get; }

        public string? City { set; get; }

        public required string ResearchAreas { set; get; }

        public string? AreaNotes { set; get; }

        public string? OrganizerWebpage { set; get; }

        public string? ContactPhoneNumber { set; get; }

        public bool IsAnonymizedFileRequired { set; get; }

        public required string AuthorInviteCode { set; get; }

        public required string ReviewerInviteCode { set; get; }

        public required string ChairInviteCode { set; get; }

        public virtual IList<Submission> Submissions { set; get; } = null!;

        public virtual IList<ApplicationUser> Participants { set; get; } = null!;

        public virtual IList<UserConferenceRole> UserRoles { set; get; } = null!;
    }
}
